core/src/main/scala/org/apache/spark/util/SizeEstimator.scala

```diff
@@ -97,6 +98,11 @@ private[spark] object SizeEstimator extends Logging {
       return System.getProperty("spark.test.useCompressedOops").toBoolean
     }
 
+    // java.vm.info provides compressed ref info for IBM JDKs
+    if (System.getProperty("java.vendor").contains("IBM")) {
```

Comment:
It would be useful to have the utility as there are a few more places coming where I'll need to know the JVM vendor.

Architecturally, are you ok with creating a public utility in launcher and having a dependency on it from core?

Feel free to commit these and I'll restructure on the fly, or I can take both back and submit multiple changes in a single pull request.